src/software/multithreading/observer.h

```diff
@@ -40,9 +40,25 @@ class Observer
      */
     virtual std::optional<T> popMostRecentlyReceivedValue(Duration max_wait_time) final;
 
-   private:
-    const size_t DEFAULT_BUFFER_SIZE = 1;
+    /**
+     * Pops the least recently received value and returns it
+     *
+     * If no value is available, this will block until:
+     * - a value becomes available
+     * - the given amount of time is exceeded
+     * - the destructor of this class is called
+     *
+     * @param max_wait_time The maximum duration to wait for a new value before
+     *                      returning
+     *
+     * @return The value least recently added to the buffer or std::nullopt if none is
+     *         available
+     */
+    virtual std::optional<T> popLeastRecentlyReceivedValue(Duration max_wait_time) final;
+
+    static constexpr size_t DEFAULT_BUFFER_SIZE = 1;
```

Comment:
DEFAULT_BUFFER_SIZE  should be private

Reply:
this is also used in `GenericThreadedObserver`, keeping it protected